### ISSUE TITLE
Poller/Consumer - potential for optimisation using iterators and combining the pollings and consuming operations

### ISSUE THREAD

**Issue by dmeehan1968** [dmeehan1968/SQSBench#6]
Splitting the poller/consumer (done with intention of making the consumer work for all poller types) has the downside that SQS works on batches of 10 messages maximum, yet the poller has to wait for an entire batch to process before it can handle deletions.  The deletions can be done in parallel, which can mean that batches of deletions up to 500 can be handled in short order.

I'm wondering if when using a manual poller it would be better to use iterators to work on the messages, so we don't have fill an array, thus overlapping some of the polling/consuming/deleting behaviours.

This might be out of scope, as whilst its a reasonable optimisation, that's always likely to be an advantage to manual polling, as one can tune to the specific requirementss.  Using whichever poller type is a way of offloading responsibility for how that works, even if its not the most optimal.

Also worth noting is that the pollers invocation isn't aligned to a minute, so even if the session duration was less than a minute, the metrics would likely still drift into the next minute.
